proposals/nnnn-vector.md

```diff
@@ -0,0 +1,846 @@
+# Vector, a fixed-size array
+
+* Proposal: [SE-NNNN](NNNN-vector.md)
+* Authors: [Alejandro Alonso](https://github.com/Azoy)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Roadmap: [Approaches for fixed-size arrays](https://forums.swift.org/t/approaches-for-fixed-size-arrays/58894)
+* Implementation: [swiftlang/swift#76438](https://github.com/swiftlang/swift/pull/76438)
+* Review: ([pitch](https://forums.swift.org/...))
+
+## Introduction
+
+This proposal introduces a new type to the standard library, `Vector`, which is
+a fixed-size array. This is analogous to the
+[classical C arrays `T[N]`](https://en.cppreference.com/w/c/language/array),
+[C++'s `std::array<T, N>`](https://en.cppreference.com/w/cpp/container/array),
+and [Rust's arrays `[T; N]`](https://doc.rust-lang.org/std/primitive.array.html).
+
+## Motivation
+
+Arrays in Swift have served as the go to choice when needing to put items in an
+ordered list. They are a great data structure ranging from a variety of
+different use cases from teaching new developers all the way up to sophisticated
+implementation details of something like a cache.
+
+However, using `Array` all the time doesn't really make sense in some scenarios.
+It's important to understand that `Array` is a heap allocated growable data
+structure which can be expensive and unnecessary in some situations. The next
+best thing is to force a known quantity of elements onto the stack, probably by
+using tuples.
+
+```swift
+func complexAlgorithm() {
+  let elements = (first, second, third, fourth)
+}
+```
+
+Unfortunately, using tuples in this way is very limited. They don't allow for
+dynamic indexing or iteration:
+
+```swift
+func complexAlgorithm() {
+  let elements = (first, second, third, fourth)
+
+  // error: for-in loop requires '(Int, Int, Int, Int)' to conform to 'Sequence'
+  for element in elements {
+    compute(element)
+  }
+
+  var i = 0
+
+  while i != elements.indices {
```

Comment:
This should be `elements.indices.count`, and we should have an `i += 1` somewhere, right?

Reply:
(Same below)

Reply:
Fixed these examples to just use a for loop over the indices.